tests/cmds/test_cases.py

```diff
@@ -0,0 +1,232 @@
+    runner.invoke(
+        cli, ["cases", "file-events", "show", "1", "--event-id", "1"], obj=cli_state,
+    )
+    cli_state.sdk.cases.file_events.get.assert_called_once_with(1, "1")
+
+
+def test_file_events_show_returns_error_message_when_py42_raises_exception(
+    runner, cli_state, error
+):
+    cli_state.sdk.cases.file_events.get.side_effect = Py42NotFoundError(error)
+    result = runner.invoke(
+        cli, ["cases", "file-events", "show", "1", "--event-id", "1"], obj=cli_state,
+    )
+    cli_state.sdk.cases.file_events.get.assert_called_once_with(1, "1")
+    assert "Invalid case-number or event-id." in result.output
+
+
+def test_file_events_add_returns_error_message_when_py42_raises_exception(
+    runner, cli_state, error
+):
+    cli_state.sdk.cases.file_events.add.side_effect = Py42BadRequestError(error)
+    result = runner.invoke(
+        cli, ["cases", "file-events", "add", "1", "--event-id", "1"], obj=cli_state,
+    )
+    cli_state.sdk.cases.file_events.add.assert_called_once_with(1, "1")
+    assert "Invalid case-number or event-id." in result.output
+
+
+def test_file_events_remove_returns_error_message_when_py42_raises_exception(
+    runner, cli_state, error
+):
+    cli_state.sdk.cases.file_events.delete.side_effect = Py42NotFoundError(error)
+    result = runner.invoke(
+        cli, ["cases", "file-events", "remove", "1", "--event-id", "1"], obj=cli_state,
+    )
+    cli_state.sdk.cases.file_events.delete.assert_called_once_with(1, "1")
+    assert "Invalid case-number or event-id." in result.output
+
+
+def test_show_returns_expected_data(runner, cli_state, py42_response):
+    py42_response.data = json.loads(CASE_DETAILS)
+    cli_state.sdk.cases.get.return_value = py42_response
+    result = runner.invoke(cli, ["cases", "show", "1"], obj=cli_state,)
+    assert "test@test.test" in result.output
+
+
+def test_list_returns_expected_data(runner, cli_state, py42_response, mocker):
+    py42_response.data = json.loads(ALL_CASES)
+
+    def gen():
+        yield py42_response.data
+
+    cli_state.sdk.cases.get_all.return_value = gen()
+    result = runner.invoke(cli, ["cases", "list"], obj=cli_state,)
+    assert "test@test.test" in result.output
+
+
+def test_events_show_returns_expected_data(runner, cli_state, py42_response):
+    py42_response.data = json.loads(EVENT_DETAILS)
+    cli_state.sdk.cases.file_events.get.return_value = py42_response
+    result = runner.invoke(cli, ["cases", "file-events", "show", "1", "--event-id", "e"], obj=cli_state,)
+    assert "0_1d71796f-af5b-4231-9d8e-df6434da4663_984418168383179707_986472527798692818_971" in result.output
+
+
+def test_events_list_returns_expected_data(runner, cli_state):
+    cli_state.sdk.cases.file_events.get_all.return_value = json.loads(ALL_EVENTS)
+    result = runner.invoke(cli, ["cases", "file-events", "list", "1"], obj=cli_state,)
+    assert "0_1d71796f-af5b-4231-9d8e-df6434da4663_984418168383179707_986472527798692818_971" in result.output
```

Comment:
Missing newline at the end of this file.

Reply:
Added.